File: frontend/app/src/types/MonthlyDeposit.ts
// railsの方の型定義と自動で合わせたいなぁ
export type MonthlyDeposit = {
  amount: number;
  order: number;
  year: number;
  simulation_id: string;
  id?: string; // フロントで保存したタイミングではid持たないのでoptional
};

File: backend/app/models/monthly_deposit.rb
class MonthlyDeposit < ApplicationRecord
  belongs_to :simulation
  validates :amount, presence: true, length: { maximum: 12 }, numericality: { only_integer: true }
  validates :year, presence: true, numericality: { only_integer: true, greater_than_or_equal_to: 1, less_than_or_equal_to: 100 }
  validates :order, presence: true, numericality: { only_integer: true, greater_than_or_equal_to: 1, less_than_or_equal_to: 100 }, uniqueness: { scope: :simulation_id }

  def self.find_by_simulation_id(id)
    joins(:simulation).where(simulations: { id: })
  end
end
